oop3-abstract-morph.js

```
@@ -9,27 +12,118 @@ Create at least 3 sub-classes that inherit from the Abstract Creature Class.  Ex
     -Reptile
     -Fish
 
-These sub-classes MUST implement the abstract methods of move() and act(), which are found on the abstract Creature class. In addition to the 2 abstract methods, implement 1 unique method on each sub-class.  Also, each sub-class MUST have at least 2 unique property fields on them.  Example property fields are:
+These sub-classes MUST implement the abstract methods of move() and act(),
+which are found on the abstract Creature class. In addition to the 2 abstract methods,
+implement 1 unique method on each sub-class.  Also, each sub-class MUST have at least 2 unique property fields on them.
+Example property fields are:
     -name
     -weight
     -food
     -age
 
 */
 
-    class Creature {
-        //your code here...
+class Creature {
+    constructor() {
+        if (this.constructor == Creature) {
+            throw new Error('Cannot instantiate Abstract Class Creature')
```

You wouldn't want to throw this error on a constructor, since child classes should call the constructor with `super()`. The reason your code works is you're checking if `this.constructor = Creature`, which is incorrect. This condition checks to see if the creature class has a field called constructor that is set equal to the Creature class, which is always false.